packages/@aws-cdk/core/lib/private/asset-staging.ts

```diff
@@ -0,0 +1,221 @@
+import { spawnSync, SpawnSyncOptions } from 'child_process';
+import * as crypto from 'crypto';
+import * as os from 'os';
+import { AssetStaging } from '../asset-staging';
+import { BundlingOptions } from '../bundling';
+
+/**
+ * Options for Docker based bundling of assets
+ */
+interface AssetStagingCopyOptions extends BundlingOptions {
+  /**
+   * Path where the source files are located
+   */
+  readonly sourcePath: string;
+  /**
+   * Path where the output files should be stored
+   */
+  readonly bundleDir: string;
+}
+
+class AssetStagingBase {
+  protected options: AssetStagingCopyOptions;
+  constructor(options: AssetStagingCopyOptions) {
+    this.options = options;
+  }
+  /**
+   * Determines a useful default user if not given otherwise
+   */
+  protected determineUser() {
+    let user: string;
+    if (this.options.user) {
+      user = this.options.user;
+    } else {
+      // Default to current user
+      const userInfo = os.userInfo();
+      user =
+        userInfo.uid !== -1 // uid is -1 on Windows
+          ? `${userInfo.uid}:${userInfo.gid}`
+          : '1000:1000';
+    }
+    return user;
+  }
+}
+
+/**
+ * Bundles files with bind mount as copy method
+ */
+export class AssetStagingBindMount extends AssetStagingBase {
+  /**
+   * Bundle files with bind mount as copy method
+   */
+  public run() {
+    this.options.image.run({
```

Comment:
I know this is currently unused but let's get ahead of ourselves.
```suggestion
    return this.options.image.run({
```